packages/client/src/Components/PaperEdits/PaperEdit/ProgramScript/PreviewCanvas/VideoContextProgressBar.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable template-curly-spacing */
+/* eslint-disable react/prop-types */
+import React from 'react';
+import styles from './VideoContextProgressBar.module.css';
+
+class VideoContextProgressBar extends React.PureComponent {
+  constructor(props) {
+    const {
+      width = 640,
+    } = props;
+
+    super(props);
+
+    this.width = width;
+    this.state = { progress: 0 };
+  }
+
+  componentDidUpdate() {
+    this.videoContext = this.props.videoContext;
+    this.setState({ duration: this.videoContext.duration });
+    this.updateProgress();
+  }
+
+  updateProgress = () => requestAnimationFrame(() => {
+    const currentTime = this.videoContext && this.videoContext.currentTime;
+    const progress = (currentTime / this.state.duration) * 100;
+
+    if (this.state.progress !== progress) this.setState({ progress });
+
+    requestAnimationFrame(this.updateProgress);
```

Comment:
Hi! I was just dropping by to look at some Videocontext stuff.

Probs a good idea to cancel this `requestAnimationFrame` before subsequent calls to `updateProgess` on line `21` and before the component unmounts.

You could just call `updateProgress` in `componentDidMount`, save a reference to the requestAnimation id on this line and then cancel it in `componentWillUnmount`?